### PR TITLE
*: clean up flags.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @c4pt0r @qiuyesuifeng @shenli 

@queenypingcap  please check the doc.

### CODE REVIEW

**Comment by queenypingcap** [comment]
LGTM

File: cmd/pd-server/main.go [hunk 1]
```diff
@@ -27,17 +28,24 @@ import (
...
-	maxPeerCount           = flag.Uint64("max-peer-count", 3, "max peer count for the region")
-	metricAddr             = flag.String("metric-addr", "", "metric address")
+	config = flag.String("c", "", "config file")
+
+	clusterID           = flag.Uint64("cluster-id", 0, "initial cluster ID for the pd cluster")
+	name                = flag.String("name", "pd", "human-readable name for this pd member")
+	dataDir             = flag.String("data-dir", "", "path to the data directory (default 'default.{name}')")
+	host                = flag.String("host", "127.0.0.1", "host for outer traffic")
+	clientPort          = flag.Uint64("client-port", 2379, "port for client traffic")
+	advertiseClientPort = flag.Uint64("advertise-client-port", 0, "advertise port for client traffic (default 2379)")
```
Comment (qiuyesuifeng) [inline cmd/pd-server/main.go]: default is 2379?

File: cmd/pd-server/main.go [hunk 2]
```diff
@@ -27,17 +28,24 @@ import (
...
+	peerPort            = flag.Uint64("peer-port", 2380, "port for peer traffic")
+	advertisePeerPort   = flag.Uint64("advertise-peer-port", 0, "advertise port for peer traffic (default 2380)")
+	port                = flag.Uint64("port", 1234, "server port (deprecate later)")
+	advertisePort       = flag.Uint64("advertise-port", 0, "advertise server port (deprecate later) (default 1234)")
+	httpPort            = flag.Uint64("http-port", 9090, "http port (deprecate later)")
+	initialCluster      = flag.String("initial-cluster", "", "initial cluster configuration for bootstrapping (default 'pd=http://127.0.0.1:2380')")
+	initialClusterState = flag.String("initial-cluster-state", "new", "initial cluster state ('new' or 'existing')")
+
+	logLevel = flag.String("L", "debug", "log level: info, debug, warn, error, fatal")
```
Comment (qiuyesuifeng) [inline cmd/pd-server/main.go]: I suggest comment will be `debug info ...`

File: cmd/pd-server/main.go (hunk 1 above)
Comment (siddontang) [inline cmd/pd-server/main.go]: yes, using 0 here is just easy for laster adjustUint64 function. 
if we set 2379 here, and use another port in clientPort and forget to change advertise port, the advertise port will be 2379, it is wrong.

File: doc/clustering.md [hunk 3]
```diff
@@ -8,46 +8,37 @@ export HostIP="192.168.199.105"
...
+          --host=${HostIP} \
           --cluster-id=1 \
-          --etcd-name=pd2 \
-          --etcd-data-dir="default.pd2" \
-          --etcd-advertise-client-url="http://${HostIP}:22379" \
-          --etcd-advertise-peer-url="http://${HostIP}:22380" \
-          --etcd-initial-cluster="pd1=http://${HostIP}:12380,pd2=http://${HostIP}:22380,pd3=http://${HostIP}:32380" \
-          --etcd-listen-peer-url="http://0.0.0.0:22380" \
-          --etcd-listen-client-url="http://0.0.0.0:22379"  
+          --name=pd2 \
```
Comment (qiuyesuifeng) [inline doc/clustering.md]: duplicated --cluster-id

File: server/config.go [hunk 4]
```diff
@@ -62,22 +64,24 @@ type Config struct {
...
 	MaxPeerCount uint64 `toml:"max-peer-count" json:"max-peer-count"`
 
-	// Remote metric address for StatsD.
-	MetricAddr string `toml:"metric-addr" json:"metric-addr"`
-
 	BalanceCfg *BalanceConfig `toml:"balance" json:"balance"`
 
+	// Server advertise listening address for outer client communication.
+	// Host:Port
+	AdvertiseAddr string
```
Comment (qiuyesuifeng) [inline server/config.go]: `toml:"advertise-addr"json:"advertise-addr"`

File: server/server.go [hunk 5]
```diff
@@ -78,9 +79,11 @@ type Server struct {
 // NewServer creates the pd server with given configuration.
 func NewServer(cfg *Config) (*Server, error) {
 	cfg.adjust()
+	log.Infof("PD config - %v", cfg)
```
Comment (qiuyesuifeng) [inline server/server.go]: In main.go, we have already had a same log, like `log.Infof("PD init config - %v", cfg)`, so should we remove that one?

File: server/config.go (hunk 4 above)
Comment (siddontang) [inline server/config.go]: it is not in config, need ?

File: server/server.go (hunk 5 above)
Comment (siddontang) [inline server/server.go]: already remove

**Comment by c4pt0r** [comment]
What's wrong with CI?

**Comment by c4pt0r** [comment]
LGTM

**Comment by c4pt0r** [comment]
Don't forget the docs.

**Pull request merged**